examples/mean-reversion.go

```diff
@@ -0,0 +1,255 @@
+		[]float64{},
+		0.0,
+		"",
+		0.0,
+		"AAPL",
+	}
+}
+
+func main() {
+	// First, cancel any existing orders so they don't impact our buying power.
+	status, until, limit := "open", time.Now(), 100
+	orders, _ := alpacaClient.client.ListOrders(&status, &until, &limit)
+	for _, order := range orders {
+		_ = alpacaClient.client.CancelOrder(order.ID)
+	}
+
+	// Wait for market to open
+	fmt.Println("Waiting for market to open...")
+	for {
+		isOpen := alpacaClient.awaitMarketOpen()
+		if isOpen {
+			break
+		}
+		time.Sleep(2000 * time.Millisecond)
+	}
+	fmt.Println("Market Opened.")
+
+	// Get the running average of prices of the last 20 minutes, waiting until we have 20 bars from market open.
+	fmt.Println("Waiting for 20 bars...")
+	for {
+		layout := "2006-01-02T15:04:05.000Z"
+		rawTime, _ := time.Parse(layout, time.Now().String())
+		currTime := rawTime.String()
+		cal, _ := alpacaClient.client.GetCalendar(&currTime, &currTime)
+		marketOpen, _ := time.Parse(layout, cal[0].Open)
+		bars, _ := alpacaClient.client.GetSymbolBars(alpacaClient.stock, alpaca.ListBarParams{Timeframe: "minute", StartDt: &marketOpen})
+		if len(bars) >= 20 {
+			break
+		} else {
+			time.Sleep(60000 * time.Millisecond)
+		}
+	}
+	fmt.Println("We have 20 bars.")
+
+	amtBars := 20
+	bars, err := alpacaClient.client.GetSymbolBars(alpacaClient.stock, alpaca.ListBarParams{Timeframe: "minute", Limit: &amtBars})
+	if err != nil {
+		fmt.Println(err)
+	}
+	for _, bar := range bars {
+		alpacaClient.closingPrices = append(alpacaClient.closingPrices, float64(bar.Close))
+		if len(alpacaClient.closingPrices) > 20 {
+			alpacaClient.closingPrices = alpacaClient.closingPrices[1:]
+		}
+		alpacaClient.runningAverage = ((alpacaClient.runningAverage * float64(len(alpacaClient.closingPrices)-1)) + float64(bar.Close)) / float64(len(alpacaClient.closingPrices))
+	}
+
+	for {
+		alpacaClient.run()
+	}
+}
+
+// Rebalance our portfolio every minute based off running average data.
+func (alp alpacaClientContainer) run() {
+	if alpacaClient.lastOrder != "" {
+		_ = alp.client.CancelOrder(alpacaClient.lastOrder)
+	}
+
+	// Rebalance the portfolio.
+	alp.rebalance()
+
+	// Figure out when the market will close so we can prepare to sell beforehand.
+	clock, _ := alp.client.GetClock()
+	timeToClose := int((clock.NextClose.UnixNano() - clock.Timestamp.UnixNano()) / 1000000)
+	if timeToClose < 60000*15 {
+		// Close all positions when 15 minutes til market close.
+		fmt.Println("Market closing soon.  Closing positions.")
+
+		positions, _ := alp.client.ListPositions()
+		for _, position := range positions {
+			var orderSide string
+			if position.Side == "long" {
+				orderSide = "sell"
+			} else {
+				orderSide = "buy"
+			}
+			qty, _ := position.Qty.Float64()
+			qty = math.Abs(qty)
+			alp.submitMarketOrder(int(qty), position.Symbol, orderSide)
+			// Run script again after market close for next trading day.
+			time.Sleep((60000 * 15) * time.Millisecond)
```

Comment:
This line can be removed

Reply:
This is done so that the function doesn't print "Market closing soon.  Closing positions" every minute, should I still remove?.  I did move it out of the for loop though, it wasn't supposed to be in there

Reply:
As long as it's not in the for loop it's fine.